runtime/onert/core/src/exec/ExecutorBase.cc

```diff
@@ -73,29 +56,29 @@ void ExecutorBase::execute(const std::vector<backend::ITensor *> &src_tensors,
 
   assert(src_tensors.size() == _graph.getInputs().size());
   assert(src_tensors.size() == _input_tensors.size());
-  for (uint32_t n = 0; n < _graph.getInputs().size(); ++n)
+  for (uint32_t n = 0; n < src_tensors.size(); ++n)
   {
     // when user changes input shape, the input tensor is dynamic and its memory is not allocated.
     // This code find the info to allocate dynamic tensor, and allocate memory based on the source
     // tensor's shape set by caller.
     const auto src_tensor = src_tensors[n];
+    assert(src_tensor->buffer() != nullptr);
     auto input_tensor = _input_tensors[n];
-    // If src_tensor or input_tensor is nullptr, pre_fn does not copy the tensors
-    if (src_tensor != nullptr && input_tensor != nullptr)
-    {
-      const auto orig_input_shape = input_tensor->getShape();
-      const auto changed_input_shape =
-          convertShape(src_tensor->getShape(), src_tensor->layout(), input_tensor->layout());
-      if (orig_input_shape != changed_input_shape)
-      {
-        input_tensor->set_dynamic();
-      }
-    }
+    assert(input_tensor != nullptr);
+    input_tensor->set_dynamic();
```

Comment:
Previous code call `set_dynamic` only when "src shape != subgraph input shape"
Shouldn't we need such check here?

Reply:
I may have removed it for testing, and forgot. I just put it back.

Reply:
Turns out it does not work, need some time to fix!

Reply:
It's hard to follow which one is fixed. :crying_cat_face:  I guess this fix is done.

Reply:
Yes it's back - https://github.com/Samsung/ONE/pull/5010/commits/f878a1313671a3f106dcab7a5de238ec20bfe00f